product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -114,3 +105,66 @@ def price_compute(self, price_type, uom=False, currency=False, company=False):
         return super(ProductProduct, self).price_compute(
             price_type, uom, currency, company
         )
+
+    def _prepare_customerinfo(self, params):
+        # This search is made to avoid retrieving from the cache.
+        return (
+            self.env["product.customerinfo"]
+            .search(
+                [
+                    ("product_tmpl_id", "=", self.product_tmpl_id.id),
+                    ("name.active", "=", True),
+                ]
+            )
+            .sorted(lambda s: (s.sequence, -s.min_qty, s.price, s.id))
+        )
+
+    def _select_customerinfo(
+        self, partner_id=False, quantity=0.0, date=None, uom_id=False, params=False
+    ):
+        """Customer version of the standard `_select_seller`.
+        """
+        self.ensure_one()
+        if date is None:
+            date = fields.Date.context_today(self)
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+
+        res = self.env["product.customerinfo"]
+        customers = self._prepare_customerinfo(params)
+        if self.env.context.get("force_company"):
+            customers = customers.filtered(
+                lambda s: not s.company_id
+                or s.company_id.id == self.env.context["force_company"]
+            )
+        for customer in customers:
+            quantity_uom_customer = quantity
+            if quantity_uom_customer and uom_id and uom_id != customer.product_uom:
+                quantity_uom_customer = uom_id._compute_quantity(
+                    quantity_uom_customer, customer.product_uom
+                )
+
+            if customer.date_start and customer.date_start > date:
+                continue
+            if customer.date_end and customer.date_end < date:
+                continue
+            if (
+                float_compare(
+                    quantity_uom_customer, customer.min_qty, precision_digits=precision
+                )
+                == -1
+            ):
+                continue
+            if partner_id and customer.name not in [partner_id, partner_id.parent_id]:
+                continue
+            if customer.product_id and customer.product_id != self:
+                continue
+            if not res or res.name == customer.name:
+                res |= customer
+        # NOTE: This differs from the standard sellers logic. This is done to
+        # keep the old behavior of this module.
+        variant_specific = res.filtered(lambda r: r.product_id)
+        if variant_specific:
+            res = variant_specific
+        return res.sorted("price")[:1]
```

Comment:
You are not preparing a hook, but changing more things here, and that should be properly documented and tested. I think we should start with the basic one first.

Reply:
I did this to keep the previous behavior in with a variant specific customerinfo has preference over a non-specific one. This is used to get the price. What do you suggest then? that I isolate this "patch" into a hook?

Reply:
No, I see the diff, and you are replacing a simple search by a lot of code, and that's why I don't get.

Reply:
@pedrobaeza Something like this?
```python
...
        return self._select_customerinfo_hook(res)

    @api.model
    def _select_customerinfo_hook(self, customers):
        # NOTE: This differs from the standard sellers logic. This is done to
        # keep the old behavior of this module.
        variant_specific = customers.filtered(lambda r: r.product_id)
        if variant_specific:
            customers = variant_specific
        return customers.sorted("price")[:1]

```

Reply:
![Selección_010](https://user-images.githubusercontent.com/7165771/126633993-b2c66cfa-8141-4f3a-b143-6dfc0ad2c980.png)

Reply:
ah, I see, maybe I should have put a link to the standard "brother": https://github.com/odoo/odoo/blob/14.0/addons/product/models/product.py#L602.

I'm doing the same Odoo does with supplieinfos but for partners, which makes more sense IMO.

Reply:
And that's what I say that should go into another PR to be discussed, as this can have side effects, changing the selector for the proper customer info on existing installations.

Reply:
why not discuss it in this one? I pinged you because that was also my concern, if you have installations with this module it can affect you. But I tried to keep the behavior the same, do you see any change that can affect your installations?

Reply:
Is there any reason for you to change the current behavior?

Anyways, if changed in this one, some tests for each edge case should be introduced, and compare before and after to be able to see.

About our current customers using this module, @sergio-teruel and @carlosdauden can say better. @chienandalu do you TCR cases?

Reply:
I think adding a method to get customerinfo is almost mandatory, I was reviewing child modules of this one and each of them searches the customerinfo in a different way... 

Once I was doing it, I thought it was good to it good, and for me the best way is to mimic Odoo; I think is an UX improvement if Odoo searches customerinfo and supplierinfo the same way. But his is optional, if you see issues with this I will prepare an alternative.

Reply:
Alternative: https://github.com/OCA/product-attribute/pull/891.